src/cloud/docker/docker-extend.md

```diff
@@ -82,26 +84,147 @@ In addition to built-in extensions, you can add PHP extensions to the PHP contai
 {:.procedure}
 To add a new PHP extension:
 
-1. Clone the `{{site.data.var.mcd}}` project to your local environment if necessary.
+1. Clone the `{{site.data.var.mcd-package}}` project to your local environment if necessary.
 
 1. On the command line, navigate to the PHP directory.
 
-   ```bash
-   cd magento-cloud-docker/src/Compose/Php
+  ```shell script
+  cd magento-cloud-docker/src/Compose/Php
+  ```
 
 1. Open the `ExtensionResolver.php` file, define the required extension inside the `getConfig` method by specifying the extension type and dependency.
 
-   For instance the following block adds the GNUPG extension:
-
-   ```php?start_inline=1
+   We have divided the extensions into three conditional groups:
+   1. `EXTENSION_TYPE_CORE` - Extension that exists in the `docker-php-source` More details [https://hub.docker.com/_/php][PHP, Docker Official Images], the part "How to install more PHP extensions"
+   1. `EXTENSION_TYPE_PECL` - Extensions that can be installed from [https://pecl.php.net/][PECL]. More details [https://hub.docker.com/_/php][PHP, Docker Official Images], the part "PECL extensions"
+   1. `EXTENSION_TYPE_INSTALLATION_SCRIPT` - For extensions that can be installed by running a number commands. More details [https://hub.docker.com/_/php][PHP, Docker Official Images], part "Other extensions"
+
+   For instance, the following block adds the `bcmath` extension:
+   
+  ```php?start_inline=1
+   'bcmath' => [
+       '>=7.0' => [self::EXTENSION_TYPE => self::EXTENSION_TYPE_CORE],
+   ],
+  ```
+  
+  In this case `bcmath` extensions will be install from `docker-php-source` for all `PHP` versions starting from `7.0`.
+   
+  In case of installing an extension from `PECL` which has a dependency on a specific package:
+  ```php?start_inline=1
    'gnupg' => [
      '>=7.0' => [
-     self::EXTENSION_TYPE => self::EXTENSION_TYPE_PECL,
-     self::EXTENSION_OS_DEPENDENCIES => ['libgpgme11-dev'],
-   ],],
-   ```
-
-   The extension type can be either CORE or PECL, which are defined as `EXTENSION_TYPE_PECL` or `EXTENSION_TYPE_CORE` respectively.
+       self::EXTENSION_TYPE => self::EXTENSION_TYPE_PECL,
+       self::EXTENSION_OS_DEPENDENCIES => ['libgpgme11-dev'],
+     ],
+   ],
+  ```
+  For extension types `EXTENSION_TYPE_PECL`\\`EXTENSION_TYPE_PECL`, the following configuration pattern is valid:
+   
+  ```php?start_inline=1
+   '<extension name>' => [ // this name will be used to identify the extension among other PHP extensions.
+       '<php version constraint>' => [ // for which PHP versions this config will apply
+            self::EXTENSION_TYPE => self::EXTENSION_TYPE_CORE // or self::EXTENSION_TYPE_PECL
+            self::EXTENSION_OS_DEPENDENCIES => ['<name dependency package 1>', <name dependency package 2>, ... ] // Linux packages, they will be installed in the order of indication before extantion
```

Comment:
```suggestion
  ```php?start_inline=1
   '<extension name>' => [ // Name to identify the extension among other PHP extensions.
       '<php version constraint>' => [ // Specifies which PHP versions this configuration applies to
            self::EXTENSION_TYPE => self::EXTENSION_TYPE_CORE // or self::EXTENSION_TYPE_PECL
            self::EXTENSION_OS_DEPENDENCIES => ['<name dependency package 1>', <name dependency package 2>, ... ] // Specifies Linux package dependencies. Thise packages install in the order listed before installing the extension
```